ArjunaJTS/jts/classes/com/arjuna/ats/internal/jts/orbspecific/coordinator/ArjunaTransactionImple.java

```diff
@@ -500,6 +500,9 @@ public void rollback () throws SystemException
 			destroyAction();
 
 			status = super.status();
+		} else if ((status == ActionStatus.ABORTING)
+				|| (status == ActionStatus.ABORTED) || (status == ActionStatus.H_ROLLBACK)) {
+			return;
```

Comment:
If you agree can you check that the TerminationTest below outputs the info message.